make/autoconf/jdk-options.m4

```diff
@@ -722,9 +722,12 @@ AC_DEFUN([JDKOPT_ENABLE_DISABLE_CDS_ARCHIVE_COH],
              test "x$OPENJDK_TARGET_CPU" != "xs390x"; then
           AC_MSG_RESULT([no (compact object headers not supported for this platform)])
           AVAILABLE=false
-        else
+        elif test "x$enable_cds_archive_coh" = "xyes"; then
           AC_MSG_RESULT([yes])
           AVAILABLE=true
+        else
+          AC_MSG_RESULT([yes])
+          AVAILABLE=false
```

Comment:
This does not look correct. The `CHECK_AVAILABLE` block is meant to check if the preconditions for enabling this feature are met. Those preconditions should not change here. All you should need to do is change the default.

Reply:
I've reverted the changes to jdk-optinos.m4.